rfq/order.go

```diff
@@ -262,7 +260,13 @@ func (c *AssetPurchasePolicy) CheckHtlcCompliance(
 			"accepted_quote_id=%v)", htlc, c.AcceptedQuoteId)
 	}
 
-	inboundAmountMSat := lnwire.MilliSatoshi(c.AssetAmount) * c.BidPrice
+	// Convert the inbound asset amount to millisatoshis and ensure that the
+	// outgoing HTLC amount is not more than the inbound asset amount.
+	assetAmt := rfqmsg.NewBigIntFixedPoint(c.AssetAmount, 0)
```

Comment:
I was wrong in what I said offline yesterday. We need the decimal display here, 0 would be wrong (see similar comments in part 1).